subsys/net/ip/tcp.c

```diff
@@ -1795,7 +1836,17 @@ static bool tcp_data_received(struct tcp *conn, struct net_pkt *pkt,
 
 	net_stats_update_tcp_seg_recv(conn->iface);
 	conn_ack(conn, *len);
-	tcp_out(conn, ACK);
+
+	/* Delay ACK response in case of small window or missing PSH,
+	 * as described in RFC 813.
+	 */
+	if (tcp_short_window(conn) || !psh) {
```

Comment:
By using the PSH symbol, you start implementing delayed ACKS see: https://en.wikipedia.org/wiki/TCP_delayed_acknowledgment. This is not without its drawbacks. It can negatively affect the functioning of the Nagle algorithm. Secondly it increases the reaction time of the 3 duplicate acks fast retransmit. Since the stack does not impact SACK, and usually is quite memory constraint, I would recommend staying away from these trouble and usually just alsways send and ack

Reply:
Ok, thanks for pointing out potential issues with delayed ACKs. I think I'll drop that part from the PR and maybe implement in separately, as a configurable option as you suggest.